include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -28,12 +28,69 @@
 #include <seqan3/range/view/to_char.hpp>
 #include <seqan3/std/ranges>
 
-namespace seqan3
+// ---------------------------------------------------------------------------------------------------------------------
+// unaligned_seq transformation trait
+// ---------------------------------------------------------------------------------------------------------------------
+
+namespace seqan3::detail
+{
+
+//!\brief Helper function to deduct the unaligned sequence type from an aligned sequence container.
+template <template <typename...> typename container_type, typename seq_alph_t, typename ...rest_t>
+//!\cond
+    requires container_concept<container_type<gapped<seq_alph_t>, rest_t...>>
+//!\endcond
+auto get_ungapped_type(container_type<gapped<seq_alph_t>, rest_t...>)
+    -> container_type<seq_alph_t, rest_t...>;
+
+//!\overload
+template <template <typename...> typename container_type,
+          template <typename...> typename allocator_type,
+          typename seq_alph_t>
+//!\cond
+    requires container_concept<container_type<gapped<seq_alph_t>, allocator_type<gapped<seq_alph_t>>>>
+//!\endcond
+auto get_ungapped_type(container_type<gapped<seq_alph_t>, allocator_type<gapped<seq_alph_t>>>)
+    -> container_type<seq_alph_t, allocator_type<seq_alph_t>>;
+
+//!\brief Default transformation trait that shall expose the unaligned sequence type of t when specialised.
+template <typename t>
+struct unaligned_seq
+{};
+
+//!\brief Exposes the unaligned sequence type given an aligned sequence container type.
+template <typename t>
+//!\cond
+    requires container_concept<t>
```

Comment:
better be paranoid and remove_reference here as well.

Reply:
and it should not have a unaligned_seq_type member type. Otherwise the requirements might lead to an ambiguous overload if there is a container with this member type.

Reply:
a gap_decorator will never model the container_concept.. but let's be paranoid

Reply:
but this does not mean that there could be an l-value reference to a container, does it? So we assume that container_concept strips references, but we should be paranoid and not assume this.